pytest-slc/exasol/pytest_slc/__init__.py

```diff
@@ -1,25 +1,80 @@
 from __future__ import annotations
-from typing import Callable
 from pathlib import Path
+from exasol.pytest_backend import paralleltask
 import pytest
 
 import pyexasol
 import exasol.bucketfs as bfs
 from exasol.python_extension_common.deployment.language_container_deployer import LanguageContainerDeployer
+from exasol.python_extension_common.deployment.language_container_builder import LanguageContainerBuilder
 
+BFS_CONTAINER_DIRECTORY = 'container'
 
-@pytest.fixture(scope="session")
-def upload_slc(backend_aware_database_params,
-               backend_aware_bucketfs_params) -> Callable[[str | Path, str, str], None]:
 
-    def func(container_file_path: str | Path,
-             language_alias: str,
-             bucketfs_path: str = ''):
+@pytest.fixture(scope='session')
+def slc_builder() -> LanguageContainerBuilder | None:
+    """
+    This fixture must be provided by the user. Below is a simple implementation.
+
+    @pytest.fixture(scope='session')
+    def slc_builder():
+        with language_container_factory() as container_builder:
+            yield container_builder
+
+    Here the language_container_factory is a context-manager function that creates a
+    LanguageContainerBuilder object. This function performs all necessary preparations for
+    building a language container, but does not call its export function. The idea is to share
+    the code that builds a language container for the release and for integration tests.
+    """
+    return None
+
+
+@pytest.fixture(scope='session', autouse=True)
+def export_slc_async(slc_builder, use_onprem: bool, use_saas: bool):
+    """
+    The fixture starts the export() function of the provided LanguageContainerBuilder object as
+    an asynchronous task.
+
+    The operation will be skipped if none of the backends is in use.
+    """
+    if (not (use_onprem or use_saas)) or (slc_builder is None):
```

Comment:
Maybe it makes also sense to add a cli option to enable building specifically also when no backend is enabled for tests that only inspect the tar archive. But, I think we can add this also later, when we see the need for it.

Reply:
PEC needs to test some general things. I meant more, if a extension needs add special things to the language container and wants to check if they are where they should be. But as we agree we can figure that out later.